packages/store/src/StoreCore.sol

```diff
@@ -214,6 +214,21 @@ library StoreCore {
       revert IStoreErrors.Store_TableAlreadyExists(tableId, string(abi.encodePacked(tableId)));
     }
 
+    // Verify that there is no table or offchain table with the same name
+    if (tableId.getType() == RESOURCE_TABLE) {
+      ResourceId tableIdOffchain = ResourceIdLib.encode(RESOURCE_OFFCHAIN_TABLE, tableId.getNameStore());
+
+      if (ResourceIds._getExists(tableIdOffchain)) {
+        revert IStoreErrors.Store_TableAlreadyExists(tableId, string(abi.encodePacked(tableId)));
+      }
+    } else if (tableId.getType() == RESOURCE_OFFCHAIN_TABLE) {
+      ResourceId tableIdOnchain = ResourceIdLib.encode(RESOURCE_TABLE, tableId.getNameStore());
+
+      if (ResourceIds._getExists(tableIdOnchain)) {
+        revert IStoreErrors.Store_TableAlreadyExists(tableId, string(abi.encodePacked(tableId)));
+      }
+    }
```

Comment:
how much worse on gas would this be if we did something like

```solidity
{
  ResourceId onchainTableId = ResourceIdLib.encode(RESOURCE_TABLE, tableId.getNameStore());
  ResourceId offchainTableId = ResourceIdLib.encode(RESOURCE_OFFCHAIN_TABLE, tableId.getNameStore());
  if (ResourceIds._getExists(onchainTableId) || ResourceIds._getExists(offchainTableId)) {
    revert IStoreErrors.Store_TableAlreadyExists(tableId, string(abi.encodePacked(tableId)));
  }
}
```

Reply:
It's actually cheaper! I guess because we do two `_getExists` checks regardless.

Much clearer too.